satpy/tests/test_utils.py

```diff
@@ -278,13 +278,18 @@ def test_specific_check_satpy(self):
         """Test 'check_satpy' with specific features provided."""
         from satpy.utils import check_satpy
         with mock.patch("satpy.utils.print") as print_mock:
-            check_satpy(readers=["viirs_sdr"], extras=("cartopy", "__fake"))
-            checked_fake = False
-            for call in print_mock.mock_calls:
-                if len(call[1]) > 0 and "__fake" in call[1][0]:
-                    assert "ok" not in call[1][1]
-                    checked_fake = True
-            assert checked_fake, "Did not find __fake module mentioned in checks"
+            check_satpy(readers=["viirs_sdr"], packages=("cartopy", "__fake"))
+            checked_fake = any("__fake: not installed" in c[1] for c in print_mock.mock_calls if len(c[1]))
+            assert checked_fake, "Did not find __fake package mentioned in checks"
+
+
+class TestShowVersions:
+    """Test the 'show_versions' function."""
+
+    def test_basic_show_versions(self):
+        """Test 'check_satpy' basic functionality."""
+        from satpy.utils import show_versions
+        show_versions()
```

Comment:
I think it would be nice to add a test with a package that doesn't exist, `show_versions(packages=["foo"])` and check that the output contains "not installed".

Reply:
this is somewhat implicitly done in the test for check_satpy that has both a missing and installed package. I think it does make sense to explicitly test it so I've added two tests.

Reply:
Oh, I missed that. Thanks for adding the extra tests! Just curious: Would it be possible to use the [capsys fixture](https://docs.pytest.org/en/stable/reference/reference.html#std-fixture-capsys) instead of mocking the print method?

Reply:
that is much cleaner, I've changed the test in this MR. perhaps in a different MR i can apply it to other places where print is being patched?

Reply:
> I've changed the test in this MR

Awesome, thanks!

> perhaps in a different MR i can apply it to other places where print is being patched

Sure, if you have some spare time, go for it :)